app/models/assets/import.rb

```diff
@@ -17,42 +17,34 @@ def json_for_remote(remote_asset)
       # Having a :to_json method that returns a json would be pretty sensible too
 
       # FOR A PLATE
-      if remote_asset.respond_to?(:wells) && remote_asset.wells
+      if remote_asset.try(:wells)
         # wells.to_a because wells relation does not act as an array
         wells = remote_asset.wells.to_a
-        if wells
-          # aliquots.to_a, same reason
-          aliquots = wells.compact.map(&:aliquots).map(&:to_a)
-          if aliquots
-            samples = aliquots.flatten.compact.map { |al| al.sample }
-            distinct += samples.compact.map(&:attributes).to_json if samples
-          end
-        end
+
+        # aliquots.to_a, same reason
+        aliquots = wells.compact.map(&:aliquots).map(&:to_a)
+        samples = aliquots.flatten.compact.map { |al| al.sample }
+        distinct += samples.compact.map(&:attributes).to_json
```

Comment:
This line was checking the condition ```if samples``` is not needed anymore?

Reply:
samples is an array, which is always truthy, even when empty, so the condition was redundant.